### PR TITLE
Feature: Prototype stream implementation

### PR DESCRIPTION
**Pull request by dangdennis**
# Request for comments
Issue #3 

- Implemented a basic read stream given input option `{ stream: true }`. 
- Todo: Need help defining `.stream` API surface without making some major changes.
- Todo: Callback function that will be given byte progress

All fetch methods can be streams, inbound and outbound network calls.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Great start! 🙌

**Comment by sindresorhus** [comment]
You are including a lot of unrelated commits. Can you properly rebase from master?

File: index.js [hunk 1]
```diff
@@ -166,6 +171,35 @@ class Ky {
...
+		let loaded = 0;
+
+		return new Response(
+			new ReadableStream({
+				start(controller) {
+					const reader = response.body.getReader();
+
+					read();
+					function read() {
+						reader.read().then(({ done, value }) => {
```
Comment (sindresorhus) [inline index.js]: Use `await` instead of `.then`

Comment (dangdennis) [reply index.js]: Will do!

1) Await
2) Fix commits

**Comment by dangdennis** [comment]
Closing this PR since I couldn't fix my the issue. Will do a new PR later with tests included as well.

**Pull request closed without merging**